scalalib/src/ScalaModule.scala

```diff
@@ -175,16 +203,24 @@ trait ScalaModule extends JavaModule { outer =>
       zincWorker.worker().docJar(
         scalaVersion(),
         scalaOrganization(),
-        scalaCompilerClasspath().map(_.path),
+        scalaDocClasspath().map(_.path),
         scalacPluginClasspath().map(_.path),
         files ++ options
       ) match{
-        case true => Result.Success(createJar(Agg(javadocDir))(outDir))
+        case true =>
+          val inputPath =
+            if (isDotty(scalaVersion())) javadocDir / '_site
+            else javadocDir
+          Result.Success(createJar(Agg(inputPath))(outDir))
         case false => Result.Failure("docJar generation failed")
       }
     }
   }
 
+  def dottydocProjectName = T { artifactName() }
+
+  def dottydocSiteRoot = T { millSourcePath / 'docs }
```

Comment:
these changes are from the original commits, but I do wonder if we should expose dotty-specific settings in the scala module? So far these are the only settings specific to dotty here.

Reply:
Well, once Dotty turns into Scala 3, we'd have to unify the settings if we want to support a single `ScalaModule` trait. And it would be odd to split them into separate traits, when it's just the next version of Scala. So it makes sense to expose the Dotty settings, though it might not be necessary in this case.

We could default to `-project ${artifactName()}` in the `scalaDocOptions` if it's a Dotty build, and maybe rename the `dottydocSiteRoot` to something more general like `docsSource` or something else if someone know a better name. The `dottydocSiteRoot` just defines where it should copy docs from, it's technically not dotty-specific, it could be used for any other doc generation tool as well which uses template files that has to be copied into the javadoc folder.

Reply:
I agree that it would be strange to split the ScalaModule trait into two. And since root and project name are actually required by dottydoc, I think it's fine to keep them as top-level defs here.

Reply:
I think, we can skip the `dottydocProjectName` and make it the default in `scalaDocOptions`. Also ScalaDoc supports setting a title, and we already support changing/customization by overriding `scalaDocOptions`.

I'm also for a rename of the `dottydocSiteRoot` target. What about `docSources` ? I'm not sure if scaladoc can include documents too, but I know that alternative JavaDoc processors like Asciidoctlet also support additional files, so it's a nice general feature.